internal/armada/server.go

```diff
@@ -330,18 +340,108 @@ func Serve(ctx context.Context, config *configuration.ArmadaConfig, healthChecks
 				return eventsPrinter.Run(ctx)
 			})
 		}
+
+		// Scheduler jobs ingester.
+		schedulerIngester := &scheduler.Ingester{
+			PulsarClient: pulsarClient,
+			ConsumerOptions: pulsar.ConsumerOptions{
+				Topic:            config.Pulsar.JobsetEventsTopic,
+				SubscriptionName: "pulsar-scheduler-ingester",
+				Type:             pulsar.KeyShared,
+			},
+			MaxWriteInterval: 10 * time.Second,
+			MaxWriteRecords:  10000,
+			Db:               pool,
```

Comment:
It looks like pool can be nil (there's a check in the deduplication code above for this) but Pulsar can be enabled.  In that case we'll pass a nil pool down to the scheduler.  How about we remove the feature flag for pulsar and assert that you have to provide postgres details.